src/module.cpp

```diff
@@ -533,11 +533,129 @@ void Module::AddGlobalVariable(const std::string &name, const Type *type, Expr *
             initExpr = TypeCheck(initExpr);
             if (initExpr != nullptr) {
                 // We need to make sure the initializer expression is
-                // the same type as the global.  (But not if it's an
-                // ExprList; they don't have types per se / can't type
-                // convert themselves anyway.)
-                if (llvm::dyn_cast<ExprList>(initExpr) == nullptr)
+                // the same type as the global.
+                if (llvm::dyn_cast<ExprList>(initExpr) == nullptr) {
                     initExpr = TypeConvertExpr(initExpr, type, "initializer");
+                } else {
+                    // The alternative is to create initializer expression with
+                    // correct type and value from ExprList.
+                    // If we have exprList that initalizes const varying int, e.g.:
+                    // static const int x = { 0, 1, 2, 3 };
+                    // then we need to convert initExpr to constExpr that can
+                    // be utilized later in arithmetic expressions that can be
+                    // calculated in compile time (Optimize function call below),
+                    ExprList *exprList = llvm::dyn_cast<ExprList>(initExpr);
+                    if (type->IsConstType() && type->IsVaryingType() && type->IsAtomicType()) {
+                        const AtomicType::BasicType basicType = CastType<AtomicType>(type)->basicType;
```

Comment:
Why going directly to `basicType` instead of calling `getBasicType()`?

Reply:
There is no such method in `AtomicType`.

Reply:
Right, I mixed it up.